### PR TITLE
[TMVA][SOFIE] Add Concat Operator

### PR DESCRIPTION
**Pull request by sanjibansg**
This pull request adds the Concat Operator in TMVA SOFIE.

- [x] Add the ROperator_Concat 
- [x] Add functionality to parse Concat operator from ONNX files
- [ ] Add functionality to parse Concat opreator from Keras' `.h5` files
- [ ] Add functionality to parse Concat operator from PyTorch's `.pt` files
- [ ] Add tests for Concat operator
   - [x] ONNX tests
   - [ ] PyTorch Tests
   - [ ] keras tests

### CODE REVIEW

**Comment by lmoneta** [comment]
@sanjibansg I think we can merge the parsing of the Concat operator and add the direct Keras and PyTorch parsing later. Do you have time to fix these conflicts and rebase ?

**Comment by sanjibansg** [comment]
> @sanjibansg I think we can merge the parsing of the Concat operator and add the direct Keras and PyTorch parsing later. Do you have time to fix these conflicts and rebase ?

Yes surely, I will rebase to fix the conflicts. I think I need to add a test for parsing the ONNX Concat Op with axis=0, I will add that as well soon.

**Comment by sanjibansg** [comment]
Closing the PR since Concat op was added by PR #10315
